File: src/main/java/no/nav/syfo/kafka/producer/OversikthendelseProducer.java
package no.nav.syfo.kafka.producer;

import no.nav.syfo.kafka.producer.model.KOversikthendelse;
import org.slf4j.Logger;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Component;

import static java.util.UUID.randomUUID;
import static org.slf4j.LoggerFactory.getLogger;

@Component
public class OversikthendelseProducer {

    private static final Logger log = getLogger(OversikthendelseProducer.class);

    private KafkaTemplate<String, Object> kafkaTemplate;

    public static final String OVERSIKTHENDELSE_TOPIC = "aapen-syfo-oversikthendelse-v1";

    public OversikthendelseProducer(
            KafkaTemplate<String, Object> kafkaTemplate
    ) {
        this.kafkaTemplate = kafkaTemplate;
    }

    public void sendOversikthendelse(KOversikthendelse kOversikthendelse) {
        try {
            kafkaTemplate.send(
                    OVERSIKTHENDELSE_TOPIC,
                    randomUUID().toString(),
                    kOversikthendelse
            ).get();
            log.info("Legger oversikthendelse med id {} på kø for enhet {}", kOversikthendelse.hendelseId(), kOversikthendelse.enhetId());
        } catch (Exception e) {
            log.error("Feil ved sending av oppgavevarsel", e);
            throw new RuntimeException("Feil ved sending av oppgavevarsel", e);
        }
    }
}

File: src/main/kotlin/no/nav/syfo/oversikthendelse/OversikthendelseProducer.kt
package no.nav.syfo.oversikthendelse

import no.nav.syfo.kafka.producer.model.KOversikthendelse
import org.slf4j.LoggerFactory
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Component
import java.util.*
import javax.inject.Inject

@Component
class OversikthendelseProducer @Inject constructor(
        private val kafkaTemplate: KafkaTemplate<String, Any>
) {
    fun sendOversikthendelse(kOversikthendelse: KOversikthendelse) {
        try {
            kafkaTemplate.send(
                    OVERSIKTHENDELSE_TOPIC,
                    UUID.randomUUID().toString(),
                    kOversikthendelse
            ).get()
            log.info("Legger oversikthendelse med id {} på kø for enhet {}", kOversikthendelse.hendelseId, kOversikthendelse.enhetId)
        } catch (e: Exception) {
            log.error("Feil ved sending av oppgavevarsel", e)
            throw RuntimeException("Feil ved sending av oppgavevarsel", e)
        }
    }

    companion object {
        private val log = LoggerFactory.getLogger(OversikthendelseProducer::class.java)
        const val OVERSIKTHENDELSE_TOPIC = "aapen-syfo-oversikthendelse-v1"
    }
}
